consensus/core/src/core.rs

```diff
@@ -39,8 +40,15 @@ pub(crate) struct Core {
     /// The block manager which is responsible for keeping track of the DAG dependencies when processing new blocks
     /// and accept them or suspend if we are missing their causal history
     block_manager: BlockManager,
-    // The commit observer is responsible for observing the commits and collecting
-    // + sending subdags over the consensus output channel.
+    /// Used to make commit decisions for leader blocks in the dag.
+    committer: UniversalCommitter,
+    /// The last decided leader returned from the universal committer. Important to note
+    /// that this does not signify that the leader has been committed yet as it still has
```

Comment:
```suggestion
    /// that this does not signify that the leader has been persisted yet as it still has
```

Reply:
The potential confusion is that `commit` is used for describing both the output of consensus, and the status of processing. I suggest using another word, e.g. persisted, to describe saving data to storage.